sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,244 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements adapter for Darts models."""
+import abc
+from typing import List, Optional
+
+import pandas
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+__author__ = ["yarnabrina"]
+
+
+class _DartsAdapter(BaseForecaster):
+    """Base adapter class for Darts models.
+
+    Parameters
+    ----------
+    past_covariates : Optional[List[str]], optional
+        column names in ``X`` which are known only for historical data, by default None
+    num_samples : Optional[int], optional
+        Number of times a prediction is sampled from a probabilistic model, by default
+        1000
+
+    Notes
+    -----
+    If unspecified, all columns will be assumed to be known during prediction duration.
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "enforce_index_type": pandas.DatetimeIndex,
+        "handles-missing-data": False,
+        "capability:insample": False,
+        "python_version": ">=3.8",
+        "python_dependencies": ["u8darts"],
+        "python_dependencies_alias": {"u8darts": "darts"},
+    }
+
+    def __init__(
+        self: "_DartsAdapter",
+        past_covariates: Optional[List[str]] = None,
+        num_samples: Optional[int] = 1000,
+    ) -> None:
+        super().__init__()
+
+        self.past_covariates = [] if past_covariates is None else past_covariates
```

Comment:
this line would violate the sklearn contract, `self.paramname` should always have same value as the `__init__` arg `paramname`

Reply:
Also, I've found another issue that probabilistic capability is probably dependent on algorithm and passed configuration. Like `XGBModel` is capable of doing so, but not unless you pass `likelihood` parameter. Is it okay to set tags based on passed arguments values? Will it cause any problem if `predict_quantiles` is expected to work only in few cases and not always?

Reply:
> Is it okay to set tags based on passed arguments values?

Yes, for that you need to do a `set_tags` after the `super().__init__` call in `__init__`.

Reply:
> Can you please suggest how will you approach this?

Use an attribute with a different name.

That is, internally, you use `self._paramname`. For a default, you set `paramname=None`, and you do `if paramname = None: self_paramname = []` or similar.

>Also, I'll probably get time for this again in the weekend, so if you can send me any reference for sklearn contract designs, I can read it up as it's coming again and again as part of different threads.

That would be here:
https://scikit-learn.org/stable/developers/develop.html

It is noteworthy that we don't follow all those conventions.

The expectations for developers/extenders are all outlined in the extension templates afaik.